File: 3주차/부끄러운어피치/괄호회전하기.java
import java.util.*;
class 괄호회전하기 {
    public int solution(String s) {
        int answer = 0;
        String str=s;
        //() [] {} 로 이루어진 괄호기호가 무작위로 배치
        //괄호 기호를 첫 입력과 똑같이 될때까지 한 문자씩 회전시킵니다
        for(int i=0;i<s.length();i++){
            char temp=str.charAt(0);
            //첫번째 문자를  맨 뒤로 보내기
            str=str.substring(1)+temp;
            if(isCorrect(str)){
                answer++;
            }
        }

        return answer;
    }
//회전시키는 도중 올바른 괄호문자열이 있으면 갯수 체크
    public boolean isCorrect (String str){
        Stack<Character> s = new Stack<>();

        char[] arr = str.toCharArray();

        for(char data:arr){
            if(s.isEmpty()){
                s.push(data);
            }
            // '(' 와 ')'  '['와 ']' 만나면 제거
            else if(s.peek()=='['&&data==']'){
                s.pop();
            }
            else if(s.peek()=='('&&data==')'){
                s.pop();
            }
            else if(s.peek()=='{'&&data=='}'){
                s.pop();
            }
            else{
                s.push(data);
            }
        }

        if(s.isEmpty()){
            return true;
        }

        return false;
    }


}

File: 3주차/박민형/괄호 회전하기.py
# 1. 회전 시키기
# 2. 회전 시킨 결과가 올바른 괄호 문자인지 판단

def solution(s):
    current_bracket = list(s)
    answer = 0
    bracket_dic = {'[': ']', '{': '}', '(': ')'}

    for _ in range(len(s)):
        stack = []
        for bracket in current_bracket:
            # 첫 문자가 여는 괄호 문자로 시작하는 경우
            if bracket == '[' or bracket == '{' or bracket == '(':
                stack.append(bracket)

            # 여는 괄호 문자가 stack에 있고 그에 상응하는 닫는 문자일 경우
            elif stack and bracket == bracket_dic[stack[-1]]:
                stack.pop()

            # stack에 아무것도 없을 때 닫힌문자로 시작하는 경우
            elif not stack and (bracket == ']' or bracket == '}' or bracket == ')'):
                answer -= 1
                break

        if not stack: answer += 1
        current_bracket = current_bracket[1:] + list(current_bracket[0])

    return answer